### PR TITLE
Updating Bucket Manager for Enabling EnableGoStorageLibrary flag

### PR DESCRIPTION
**Pull request by Tulsishah**
created bucket_manager_test, facing an error in 2nd test case, created fake storage bucket handle, made buckethandle and storageclient public for that

### CODE REVIEW

File: internal/gcsx/bucket_manager.go [hunk 1]
```diff
@@ -40,6 +42,8 @@ type BucketConfig struct {
 	StatCacheCapacity                  int
 	StatCacheTTL                       time.Duration
 	EnableMonitoring                   bool
+	EnableStorageClientLibrary         bool
+	debugGcs                           bool
```
Comment (vadlakondaswetha) [inline internal/gcsx/bucket_manager.go:46]: why is everything else public and debugGcs private?

File: internal/gcsx/bucket_manager.go [hunk 2]
```diff
@@ -73,18 +76,20 @@ type BucketManager interface {
 }
 
 type bucketManager struct {
-	config BucketConfig
-	conn   *Connection
+	config        BucketConfig
+	conn          *Connection
+	storageHandle storage.StorageHandle
```
Comment (vadlakondaswetha) [inline internal/gcsx/bucket_manager.go:81]: Please make storage.StorageHandle as pointer.

File: internal/gcsx/bucket_manager.go [hunk 3]
```diff
@@ -148,13 +153,16 @@ func setUpRateLimiting(
...
-func (bm *bucketManager) SetUpBucket(
-	ctx context.Context,
-	name string) (sb SyncerBucket, err error) {
-	var b gcs.Bucket
-	// Set up the appropriate backing bucket.
-	if name == canned.FakeBucketName {
-		b = canned.MakeFakeBucket(ctx)
+func (bm *bucketManager) SetUpGcsBucket(ctx context.Context, name string) (b gcs.Bucket, err error) {
+	if bm.config.EnableStorageClientLibrary {
+		bh, _ := bm.storageHandle.BucketHandle(name)
```
Comment (vadlakondaswetha) [inline internal/gcsx/bucket_manager.go:158]: you shouldn't ignore the error here. In case of error err should be set and return instead of wrapping with ReqTrace and DebugBucket.

File: internal/gcsx/bucket_manager_test.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+func (t *BucketManagerTest) TestSetUpGcsBucketEnableStorageClientLibraryMethod() {
+	var bm bucketManager
+	var nilBucket *storage.BucketHandle = nil
+	bm.storageHandle = &storage.Storageclient{Client: t.fakeStorageServer.Client()}
+	bm.config.EnableStorageClientLibrary = true
+
+	Bucket, err := bm.SetUpGcsBucket(context.Background(), TestBucketName)
+
+	ExpectNe(Bucket, nilBucket)
```
Comment (vadlakondaswetha) [inline internal/gcsx/bucket_manager_test.go:78]: why do you need a variable for nil value. you can simply use
ExpectNe(bucket, nil)

File: internal/gcsx/bucket_manager_test.go [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+	AssertEq(nil, err)
+	AssertNe(nil, t.bucketHandle)
+}
+
+func (t *BucketManagerTest) TearDown() {
+	t.fakeStorageServer.Stop()
+}
+
+func (t *BucketManagerTest) TestNewBucketManagerMethod() {
+	var nilValue *bucketManager = nil
```
Comment (vadlakondaswetha) [inline internal/gcsx/bucket_manager_test.go:50]: no need of variable for nil value. Please remove it everywhere.

File: internal/gcsx/bucket_manager_test.go [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+	storageClient := &storage.Storageclient{Client: t.fakeStorageServer.Client()}
+
```
Comment (vadlakondaswetha) [inline internal/gcsx/bucket_manager_test.go:52]: new lines should be between
setup
<<new line>>
test method invocation
<<new line>>
assert statement

File: internal/gcsx/bucket_manager_test.go [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+		debugGcs:                           true,
+		AppendThreshold:                    2,
+		TmpObjectPrefix:                    "TmpObjectPrefix",
+	}
+
+	bm := NewBucketManager(bucketConfig, nil, storageClient)
+	ExpectNe(bm, nilValue)
+}
+
+func (t *BucketManagerTest) TestSetUpGcsBucketEnableStorageClientLibraryMethod() {
```
Comment (vadlakondaswetha) [inline internal/gcsx/bucket_manager_test.go:70]: Rename to
TestSetupGcsBucketWhenEnableStorageClientLibraryIsTrue

File: internal/gcsx/bucket_manager_test.go [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+	bm.storageHandle = &storage.Storageclient{Client: t.fakeStorageServer.Client()}
+	bm.config.EnableStorageClientLibrary = true
+
+	Bucket, err := bm.SetUpGcsBucket(context.Background(), TestBucketName)
+
+	ExpectNe(Bucket, nilBucket)
+	ExpectEq(err, nil)
+}
+
+func (t *BucketManagerTest) TestSetUpGcsBucketDisableStorageClientLibraryMethod() {
```
Comment (vadlakondaswetha) [inline internal/gcsx/bucket_manager_test.go:82]: same as above method except whenEnableStorageClientLibraryIsFalse

File: internal/gcsx/bucket_manager_test.go [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+	ExpectNe(bm, nilValue)
+}
+
+func (t *BucketManagerTest) TestSetUpGcsBucketEnableStorageClientLibraryMethod() {
+	var bm bucketManager
+	var nilBucket *storage.BucketHandle = nil
+	bm.storageHandle = &storage.Storageclient{Client: t.fakeStorageServer.Client()}
+	bm.config.EnableStorageClientLibrary = true
+
+	Bucket, err := bm.SetUpGcsBucket(context.Background(), TestBucketName)
```
Comment (vadlakondaswetha) [inline internal/gcsx/bucket_manager_test.go:76]: is it possible to check if buckets are wrapped correctly i.e, bucket should be of type DebugBucket and wrapped variable should be of type RequestTrace and so on.

File: internal/storage/bucket_handle.go [hunk 10]
```diff
@@ -31,20 +31,20 @@ import (
 	"google.golang.org/api/googleapi"
 )
 
-type bucketHandle struct {
+type BucketHandle struct {
 	gcs.Bucket
-	bucket *storage.BucketHandle
+	BucketObj *storage.BucketHandle
```
Comment (vadlakondaswetha) [inline internal/storage/bucket_handle.go:36]: Why are these changes required. Atleast i dont see them in this PR.

**Review by vadlakondaswetha — CHANGES_REQUESTED** [review]

File: internal/gcsx/bucket_manager_test.go (hunk 4 above)
Comment (Tulsishah) [reply internal/gcsx/bucket_manager_test.go:78]: Because it was sawing type does not match error
like Bukcet type is gcs.bucket and nil has not same type

File: internal/gcsx/bucket_manager.go (hunk 1 above)
Comment (Tulsishah) [reply internal/gcsx/bucket_manager.go:46]: It was previously initialised flag, I didn't need to change it. should I update it?

File: internal/gcsx/bucket_manager.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (Tulsishah) [reply internal/gcsx/bucket_manager.go:158]: we need bh  below and assign it into b
Ex. b = gcs.GetWrappedWithReqtraceBucket(bh)

[the remark this answers is not in the document]
Comment (Tulsishah) [reply internal/gcsx/bucket_manager.go:158]: It is sawing this error
Cannot use 'gcs.GetWrappedWithReqtraceBucket(b)' (type *reqtraceBucket) as the type *BucketHandle

File: internal/gcsx/bucket_manager.go (hunk 1 above)
Comment (vadlakondaswetha) [reply internal/gcsx/bucket_manager.go:46]: But added as part of GoClient library changes onnly right. I think you need to make it public

File: internal/gcsx/bucket_manager_test.go (hunk 9 above)
Comment (Tulsishah) [reply internal/gcsx/bucket_manager_test.go:76]: No, because debugbucket and RequestTrace are unexported type of buckets from jacobsa/gcloud

File: internal/gcsx/bucket_manager.go (hunk 1 above)
Comment (Tulsishah) [reply internal/gcsx/bucket_manager.go]: fixed

File: internal/gcsx/bucket_manager.go (hunk 3 above)
Comment (Tulsishah) [reply internal/gcsx/bucket_manager.go]: fixed

File: internal/gcsx/bucket_manager_test.go (hunk 9 above)
[the remark this answers is not in the document]
Comment (Tulsishah) [reply internal/gcsx/bucket_manager_test.go]: fixed

File: internal/gcsx/bucket_manager_test.go (hunk 6 above)
Comment (Tulsishah) [reply internal/gcsx/bucket_manager_test.go]: fixed

File: internal/gcsx/bucket_manager_test.go (hunk 7 above)
Comment (Tulsishah) [reply internal/gcsx/bucket_manager_test.go]: fixed

File: internal/gcsx/bucket_manager_test.go (hunk 8 above)
Comment (Tulsishah) [reply internal/gcsx/bucket_manager_test.go]: fixed

File: internal/storage/bucket_handle.go (hunk 10 above)
Comment (Tulsishah) [reply internal/storage/bucket_handle.go]: fixed

File: internal/gcsx/bucket_manager.go (hunk 2 above)
Comment (vadlakondaswetha) [reply internal/gcsx/bucket_manager.go:81]: checked the code
connection is a struct and storageHandle is a Interface. Since interface is kind of pointer which holds the concrete instance no need to make it as pointer.

Comment (Tulsishah) [reply internal/gcsx/bucket_manager.go:81]: Method is not callable through pointer

File: internal/gcsx/bucket_manager.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (vadlakondaswetha) [reply internal/gcsx/bucket_manager.go]: this code worked for me

b, err = bm.storageHandle.BucketHandle(name)

		if reqtrace.Enabled() {
			b = gcs.GetWrappedWithReqtraceBucket(b)
		}
		if bm.config.debugGcs {
			b = gcs.NewDebugBucket(b, logger.NewDebug("gcs: "))
		}

File: internal/gcsx/bucket_manager_test.go (hunk 9 above)
Comment (vadlakondaswetha) [reply internal/gcsx/bucket_manager_test.go]: what do you mean by unexported types?

Comment (Tulsishah) [reply internal/gcsx/bucket_manager_test.go]: It is not public

File: internal/gcsx/bucket_manager.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (Tulsishah) [reply internal/gcsx/bucket_manager.go]: fixed

File: internal/gcsx/bucket_manager_test.go (hunk 4 above)
Comment (vadlakondaswetha) [reply internal/gcsx/bucket_manager_test.go]: ordering of arguments is wrong.
first parameter is expected value and 2nd param is actual value
if you order them correctly, it works.

File: internal/gcsx/bucket_manager_test.go [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+
+func (t *BucketManagerTest) SetUp(_ *TestInfo) {
+	var err error
+
+	t.fakeStorageServer, err = storage.CreateFakeStorageServer([]fakestorage.Object{storage.GetTestFakeStorageObject()})
+
+	AssertEq(nil, err)
+
+	storageClient := &storage.Storageclient{Client: t.fakeStorageServer.Client()}
+	t.bucketHandle, err = storageClient.BucketHandle(TestBucketName)
```
Comment (vadlakondaswetha) [inline internal/gcsx/bucket_manager_test.go:42]: can you assign gcs.Bucket here, so making public of bucketHandle is not required.

File: internal/gcsx/bucket_manager_test.go (hunk 5 above)
Comment (Tulsishah) [reply internal/gcsx/bucket_manager_test.go]: fixed

File: internal/gcsx/bucket_manager_test.go (hunk 4 above)
Comment (Tulsishah) [reply internal/gcsx/bucket_manager_test.go]: fixed

File: internal/gcsx/bucket_manager_test.go (hunk 11 above)
Comment (Tulsishah) [reply internal/gcsx/bucket_manager_test.go]: fixed

File: internal/gcsx/bucket_manager.go [hunk 12]
```diff
@@ -148,13 +153,19 @@ func setUpRateLimiting(
...
-	name string) (sb SyncerBucket, err error) {
-	var b gcs.Bucket
-	// Set up the appropriate backing bucket.
-	if name == canned.FakeBucketName {
-		b = canned.MakeFakeBucket(ctx)
+func (bm *bucketManager) SetUpGcsBucket(ctx context.Context, name string) (b gcs.Bucket, err error) {
+	if bm.config.EnableStorageClientLibrary {
+		b, err = bm.storageHandle.BucketHandle(name)
+		if err != nil {
+			fmt.Println("Error Occurred while handling bucket ")
```
[the remark this answers is not in the document]
Comment (Tulsishah) [reply internal/gcsx/bucket_manager.go]: fixed

File: internal/gcsx/bucket_manager.go [hunk 13]
```diff
@@ -148,13 +153,19 @@ func setUpRateLimiting(
...
-	var b gcs.Bucket
-	// Set up the appropriate backing bucket.
-	if name == canned.FakeBucketName {
-		b = canned.MakeFakeBucket(ctx)
+func (bm *bucketManager) SetUpGcsBucket(ctx context.Context, name string) (b gcs.Bucket, err error) {
+	if bm.config.EnableStorageClientLibrary {
+		b, err = bm.storageHandle.BucketHandle(name)
+		if err != nil {
+			return
+		} else {
```
Comment (vadlakondaswetha) [inline internal/gcsx/bucket_manager.go:161]: if err is not nil we are returning without executing further code. you dont need else block below. Below code will be executed only if err !=nil

File: internal/gcsx/bucket_manager_test.go [hunk 14]
```diff
@@ -0,0 +1,127 @@
+package gcsx
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/fsouza/fake-gcs-server/fakestorage"
```
Comment (vadlakondaswetha) [inline internal/gcsx/bucket_manager_test.go:8]: i would prefer not getting this dependency out of fake_storage_util.go file. Can you add a method 
StorageHandle([]fakestorage.Object{storage.GetTestFakeStorageObject()}) -> which creates a fakeStorageServer, then storageHandle instance and returns. This way we can avoid making all structs public in storageHandle class and also scope of fakeStorageServer will be limited to fake_storagE_util..


StorageHandle() {
server fakestorage.Server =  CreateFakeStorageServer(GetTestFakeStorageObject())
return &storage.Storageclient{Client: t.fakeStorageServer.Client()}
 
}

**Review by vadlakondaswetha — CHANGES_REQUESTED** [review]

File: internal/gcsx/bucket_manager.go (hunk 13 above)
Comment (Tulsishah) [reply internal/gcsx/bucket_manager.go]: fixed

File: internal/gcsx/bucket_manager_test.go (hunk 14 above)
Comment (Tulsishah) [reply internal/gcsx/bucket_manager_test.go]: fixed

File: internal/storage/fake_storage_util.go [hunk 15]
```diff
@@ -26,7 +26,20 @@ const TestObjectName string = "gcsfuse/default.txt"
 const ContentInTestObject string = "Hello GCSFuse!!!"
 const TestObjectGeneration int64 = 780
 
-func GetTestFakeStorageObject() fakestorage.Object {
+type fakeStorage interface {
```
Comment (vadlakondaswetha) [inline internal/storage/fake_storage_util.go:29]: interface should always be public. Why is interface not public and struct is public?

File: internal/gcsx/bucket_manager_test.go [hunk 16]
```diff
@@ -0,0 +1,122 @@
...
+const TestBucketName string = "gcsfuse-default-bucket"
+
+////////////////////////////////////////////////////////////////////////
+// Boilerplate
+////////////////////////////////////////////////////////////////////////
+
+type BucketManagerTest struct {
+	bucket            gcs.Bucket
+	storageHandle     *storage.Storageclient
+	fakeStorageServer storage.FakeStorage
```
Comment (vadlakondaswetha) [inline internal/gcsx/bucket_manager_test.go:26]: just call it fakeStorage instead of fakeStorageServer

File: internal/gcsx/bucket_manager_test.go [hunk 17]
```diff
@@ -0,0 +1,122 @@
...
+	AssertEq(nil, err)
+	AssertNe(nil, t.bucket)
+}
+
+func (t *BucketManagerTest) TearDown() {
+	t.fakeStorageServer.ShutDown()
+}
+
+func (t *BucketManagerTest) TestNewBucketManagerMethod() {
+	storageClient := t.storageHandle
```
Comment (vadlakondaswetha) [inline internal/gcsx/bucket_manager_test.go:48]: why are you extacting it a seperate varable, you can pass t.storageHandle directly to NewBucketManager() method.

File: internal/gcsx/bucket_manager_test.go (hunk 16 above)
Comment (vadlakondaswetha) [inline internal/gcsx/bucket_manager_test.go:26]: why do you need fakeStorageServer instance as part of BucketMangerTest struct?

File: internal/gcsx/bucket_manager.go [hunk 18]
```diff
@@ -148,13 +153,18 @@ func setUpRateLimiting(
...
-	var b gcs.Bucket
-	// Set up the appropriate backing bucket.
-	if name == canned.FakeBucketName {
-		b = canned.MakeFakeBucket(ctx)
+func (bm *bucketManager) SetUpGcsBucket(ctx context.Context, name string) (b gcs.Bucket, err error) {
+	if bm.config.EnableStorageClientLibrary {
+		b, err = bm.storageHandle.BucketHandle(name)
+		if err != nil {
+			return
+		}
```
Comment (vadlakondaswetha) [inline internal/gcsx/bucket_manager.go:161]: nit: please add new line after line 161.

File: internal/storage/storage_handle.go [hunk 19]
```diff
@@ -31,8 +31,8 @@ type StorageHandle interface {
 	BucketHandle(bucketName string) (bh *bucketHandle, err error)
 }
 
-type storageClient struct {
-	client *storage.Client
+type Storageclient struct {
```
Comment (vadlakondaswetha) [inline internal/storage/storage_handle.go:34]: do we still need this to be public

**Review by vadlakondaswetha — CHANGES_REQUESTED** [review]

File: internal/gcsx/bucket_manager_test.go (hunk 16 above)
Comment (Tulsishah) [reply internal/gcsx/bucket_manager_test.go:26]: for shutdown the server we need the fakestorage instance

File: internal/storage/storage_handle.go (hunk 19 above)
Comment (Tulsishah) [reply internal/storage/storage_handle.go:34]: Yes because we are returning storageclient object in fake_storage_util and receiving it in bucket_manager_test

File: internal/gcsx/bucket_manager_test.go (hunk 16 above)
Comment (Tulsishah) [reply internal/gcsx/bucket_manager_test.go]: fixed

File: internal/gcsx/bucket_manager_test.go (hunk 17 above)
Comment (Tulsishah) [reply internal/gcsx/bucket_manager_test.go]: fixed

File: internal/storage/fake_storage_util.go (hunk 15 above)
Comment (Tulsishah) [reply internal/storage/fake_storage_util.go]: fixed

File: internal/gcsx/bucket_manager.go (hunk 18 above)
Comment (Tulsishah) [reply internal/gcsx/bucket_manager.go:161]: fixed

File: internal/storage/storage_handle.go (hunk 19 above)
Comment (Tulsishah) [reply internal/storage/storage_handle.go]: fixed

File: internal/gcsx/bucket_manager_test.go (hunk 16 above)
Comment (Tulsishah) [reply internal/gcsx/bucket_manager_test.go]: fixed

File: internal/storage/fake_storage_util.go [hunk 20]
```diff
@@ -26,6 +26,23 @@ const TestObjectName string = "gcsfuse/default.txt"
...
+	CreateStorageHandle() (storageHandleObj StorageHandle)
+
+	ShutDown()
+}
+
+type fakeStorage struct {
+	fakeStorageServer *fakestorage.Server
+}
+
+func NewFakeStorage(f *fakestorage.Server) FakeStorageServer {
```
Comment (vadlakondaswetha) [inline internal/storage/fake_storage_util.go:39]: NewFakeStorage should create new fakestorage.server and return, instead of taking fakestorage.server as argument

File: internal/storage/fake_storage_util.go [hunk 21]
```diff
@@ -26,6 +26,23 @@ const TestObjectName string = "gcsfuse/default.txt"
 const ContentInTestObject string = "Hello GCSFuse!!!"
 const TestObjectGeneration int64 = 780
 
+type FakeStorageServer interface {
```
Comment (vadlakondaswetha) [inline internal/storage/fake_storage_util.go:29]: keep the interface and struct name same

File: internal/storage/fake_storage_util.go [hunk 22]
```diff
@@ -45,6 +62,21 @@ func CreateFakeStorageServer(objects []fakestorage.Object) (*fakestorage.Server,
 	})
 }
 
+func CreateFakeStorageServerWithTestObject() (*fakestorage.Server, error) {
```
Comment (vadlakondaswetha) [inline internal/storage/fake_storage_util.go:65]: please order the methods in a file appropriately. It will be good to keep interface methods first and then other public methods unless interface methods internally call these public methods.

File: internal/storage/fake_storage_util.go [hunk 23]
```diff
@@ -45,6 +62,21 @@ func CreateFakeStorageServer(objects []fakestorage.Object) (*fakestorage.Server,
...
+	var err error
+	fakeStorageServer, err := CreateFakeStorageServer([]fakestorage.Object{GetTestFakeStorageObject()})
+	return fakeStorageServer, err
+}
+
+func (f *fakeStorage) CreateStorageHandle() (storageHandleObj StorageHandle) {
```
Comment (vadlakondaswetha) [inline internal/storage/fake_storage_util.go:71]: nit: rename storageHandleObj to sh

**Review by vadlakondaswetha — CHANGES_REQUESTED** [review]

File: internal/storage/fake_storage_util.go (hunk 20 above)
Comment (Tulsishah) [reply internal/storage/fake_storage_util.go]: fixed

File: internal/storage/fake_storage_util.go (hunk 22 above)
Comment (Tulsishah) [reply internal/storage/fake_storage_util.go:75]: fixed

File: internal/storage/fake_storage_util.go (hunk 23 above)
Comment (Tulsishah) [reply internal/storage/fake_storage_util.go]: fixed

File: internal/storage/fake_storage_util.go (hunk 21 above)
Comment (Tulsishah) [reply internal/storage/fake_storage_util.go]: fixed

File: internal/storage/fake_storage_util.go [hunk 24]
```diff
@@ -26,6 +26,33 @@ const TestObjectName string = "gcsfuse/default.txt"
 const ContentInTestObject string = "Hello GCSFuse!!!"
 const TestObjectGeneration int64 = 780
 
+type FakeStorage interface {
+	CreateStorageHandle() (storageHandleObj StorageHandle)
```
Comment (vadlakondaswetha) [inline internal/storage/fake_storage_util.go:30]: can you rename storageHandleObj to sh here as well.

File: internal/storage/fake_storage_util.go [hunk 25]
```diff
@@ -45,6 +72,12 @@ func CreateFakeStorageServer(objects []fakestorage.Object) (*fakestorage.Server,
 	})
 }
 
+func CreateFakeStorageServerWithTestObject() (*fakestorage.Server, error) {
```
Comment (vadlakondaswetha) [inline internal/storage/fake_storage_util.go:75]: now that you have updated bucket_handle_test and storage_handle_test, 
1. do you still need CreateFakeStorageServer method
2. Does CreateFakeStorageServerWithTestObject need to be public?
3. Same for GetTestFakeStorageObject, should it be public?

File: internal/storage/storage_handle_test.go [hunk 26]
```diff
@@ -66,19 +65,15 @@ func (t *StorageHandleTest) invokeAndVerifyStorageHandle(sc StorageClientConfig)
 }
 
 func (t *StorageHandleTest) TestBucketHandleWhenBucketExists() {
-	fakeClient := t.fakeStorageServer.Client()
-	storageClient := &storageClient{client: fakeClient}
-
+	storageClient := t.fakeStorage.CreateStorageHandle()
```
Comment (vadlakondaswetha) [inline internal/storage/storage_handle_test.go:68]: rename the variable to storageHandle. You are creating storageHandle instance here not storageClient. 
Same for below test as well.

**Review by vadlakondaswetha — CHANGES_REQUESTED** [review]

File: internal/storage/fake_storage_util.go (hunk 24 above)
Comment (Tulsishah) [reply internal/storage/fake_storage_util.go]: fixed

File: internal/storage/fake_storage_util.go (hunk 25 above)
Comment (Tulsishah) [reply internal/storage/fake_storage_util.go]: fixed

File: internal/storage/storage_handle_test.go (hunk 26 above)
Comment (Tulsishah) [reply internal/storage/storage_handle_test.go]: fixed

**Review by vadlakondaswetha — APPROVED** [review]
